mmv1/api/resource.go

```diff
@@ -2109,7 +2109,13 @@ func (r Resource) ShouldDatasourceSetAnnotations() bool {
 // that should be marked as "Required".
 func (r Resource) DatasourceRequiredFields() []string {
 	requiredFields := []string{}
-	uriParts := strings.Split(r.SelfLink, "/")
+	var uri string
+	if len(r.ImportFormat) > 0 {
+		uri = r.ImportFormat[0]
+	} else {
+		uri = r.SelfLink
+	}
+	uriParts := strings.Split(uri, "/")
```

Comment:
Per the other comment, could this be changed to `uriParts := strings.Split(r.IdFormat, "/")` to simplify?